src/FffGcodeWriter.cpp

```diff
@@ -654,6 +654,19 @@ LayerPlan& FffGcodeWriter::processLayer(const SliceDataStorage& storage, int lay
             }
             layer_thickness = getSettingInMicrons("layer_height_0");
         }
+        int64_t z_offset = getSettingInMicrons("z_offset_layer_0");
```

Comment:
You should use `coord_t` here instead of `int64_t`. In practice it's the same, but the length of the coordinates could be changed later.